### ISSUE TITLE
Problem: poll-ingest-activity fails because of an i/o timeout

### ISSUE THREAD

**Issue by jorikvankemenade** [artefactual-labs/enduro#51]
**Expected**
Timeouts are gracefully handled by retrying.

**Current situation**
I have had a few cases where I have a workflow failure:

```
non retryable error - error checking ingest status (non retryable error): Get http://archivematica-dashboard-dev.cern.ch/api/ingest/status/6de105df-eebe-4189-aa62-3c3632cf1f84: dial tcp: i/o timeout
```

I am not sure what the origin of this error is, or what causes it. It is not a propper `HTTP` status error, which has a nice retry path. Currently, I think it might be directly from Go's HTTP package, but I am not sure. One thing I found is that Enduro uses the default  HTTP Client. Some sources on the internet ([random example](https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779)) suggest that this is not a good idea. So maybe this might be a solution, but I am not sure.

**Comment by sevein** [comment]
> One thing I found is that Enduro uses the default HTTP Client.

Each pipeline has its own custom HTTP client create [here](https://github.com/artefactual-labs/enduro/blob/0f84fdf2ed4eb8d03cfaca84d7271529143f7c34/internal/pipeline/pipeline.go#L149-L166). The area of the code relevant to your example is in pipeline/status.go:

```go
status, _, err := client.Ingest.Status(ctx, ID)
if err != nil {
    if errors.Is(err, context.DeadlineExceeded) {
        // return retryable err ...
    }
    if err, ok := err.(*amclient.ErrorResponse); ok {
        // return retryable/non-retryable depending on status codes ...
    }
    // return non-retryable ...
}
```

Maybe we need to add a case for `os.IsTimeout(err)`. https://golang.org/doc/go1.13#net describes something related.

**Comment by sevein** [comment]
Fixed in v0.22.2.

**Comment by jorikvankemenade** [comment]
Sorry, yesterday I busy so I wasn't able to update the issue. When I was testing the proposed fix I found that `os.IsTimeout(err)` does not catch the error. The error is wrapped inside a `net.Error`. I'll open a PR with my successfully tested changes.
